### PR TITLE
[Re-open of #6422] Introducing subAssessments type (Microsoft.Security provider) 

### PR DESCRIPTION
**Pull request by himarkov**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by himarkov** [comment]
Reopen of signed off PR (with paths change)
https://github.com/Azure/azure-rest-api-specs/pull/6422

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]
When will this pr be ready?

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/subAssessments.json [hunk 1]
```diff
@@ -0,0 +1,576 @@
...
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/{scope}/providers/Microsoft.Security/subAssessments": {
```
Comment (pilor) [inline specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/subAssessments.json]: If subAssessments are a child of assessments it doesn't make sense to get a collection of them outside of an individual assessment.  Looks like this is the only change from the original PR.  Is this a scenario you need to support?

Comment (himarkov) [reply specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/subAssessments.json]: Its necessary as theres a requirement to expose them all on subscription level (mostly for scale issues on synching this data with ARG)
The parent is reflected on the object itself nevertheless , therefore we don’t loss data on this path, its only a different journey to get to it.

**Comment by himarkov** [comment]
Hi, this pr is ready on our end

**Review by pilor — APPROVED** [review]

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/subAssessments.json (hunk 1 above)
Comment (pilor) [reply specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/subAssessments.json]: Have you tried this in dogfood?  How will you represent this type in your manifest?  as assessments/subAssessments and just "subAssessments"?  Currently your manifest PR has it just listed as a "subAssessments" resourceType which seems a bit misleading.  RBAC also becomes an issue when its reprsented as two different actions.

Did ARG specifically ask for this?  Subscription level GETs are typically only used for top-level resources.

**Comment by himarkov** [comment]
Hi,

ARG has specifically asked this, as otherwise, upon sync, they will be needed to query per each subscription + assessment id combo, which makes the query scale pretty huge for both our side and their.
Regarding the type, it remains Microsoft.security/assessments/subassessments for both APIs.

This has been tried.
Regarding the manifest pr – if im not wrong, it doesn’t has the type represented in it, but only the name and route. Have I missed something?

Thanks,
Hila

**Review by ArcturusZhang — APPROVED** [review]

**Comment by ArcturusZhang** [comment]
Hi @himarkov I approved these changes for now. When this is ready to merge, please let me know. If this is not going to get ready for short, I suggest to close this pr, and open a new one when this is ready to merge

**Comment by himarkov** [comment]
Hi, this is ready to be merged.
Thanks.

**Comment by ArcturusZhang** [comment]
> Hi, this is ready to be merged.
> Thanks.

Merged

**Pull request merged**